File: client-node/src/main/scala/pl/jaca/ircsy/clientnode/connection/messages/PrivateMessage.scala
package pl.jaca.ircsy.clientnode.connection.messages

import java.time.LocalDate

/**
  * @author Jaca777
  *         Created 2016-05-07 at 21
  */
case class PrivateMessage(firstParticipant: String, secondParticipant: String, sender: String, time: LocalDate, message: String) {

}

File: chat/src/main/java/pl/jaca/ircsy/chat/messages/PrivateMessage.java
package pl.jaca.ircsy.chat.messages;

import pl.jaca.ircsy.chat.PrivateChat;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * @author Jaca777
 *         Created 2016-05-13 at 22
 */
public class PrivateMessage implements Serializable {

    private LocalDateTime time;
    private PrivateChat chat;
    private ChatUser author;
    private String text;

    public PrivateMessage(LocalDateTime time, PrivateChat chat, ChatUser author, String text) {
        this.time = time;
        this.chat = chat;
        this.author = author;
        this.text = text;
    }

    public LocalDateTime getTime() {
        return time;
    }

    public PrivateChat getChat() {
        return chat;
    }

    public ChatUser getAuthor() {
        return author;
    }

    public String getText() {
        return text;
    }
}
